File: service/src/main/java/com/github/goldy1992/mp3player/dagger/modules/MyDrawerListenerModule.java
package com.github.goldy1992.mp3player.dagger.modules;

import com.github.goldy1992.mp3player.client.MyDrawerListener;

import dagger.Module;
import dagger.Provides;

@Module
public class MyDrawerListenerModule {

    @Provides
    public MyDrawerListener provideMyDrawerListener() {
        return new MyDrawerListener();
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/dagger/modules/MyDrawerListenerModule.kt
package com.github.goldy1992.mp3player.client.dagger.modules

import com.github.goldy1992.mp3player.client.listeners.MyDrawerListener
import dagger.Module
import dagger.Provides

@Module
class MyDrawerListenerModule {
    @Provides
    fun provideMyDrawerListener(): MyDrawerListener {
        return MyDrawerListener()
    }
}